### ISSUE TITLE
Can't list ACR credentials after upgrade to 2.64.0 (C#)

### ISSUE THREAD

**Issue by Raniz85** [pulumi/pulumi-azure-native#3608]
### What happened?

I upgraded Pulumi.AzureNative to 2.64.0 today, and now I can no longer preview or deploy my stack. Either command hangs for about 10 minutes and then exits with an exception.

If I downgrade to 2.63.0 again, it works.

### Example

```csharp
        Registry = new Registry("registry", new RegistryArgs
        {
            ResourceGroupName = resourceGroup.Name,
            Sku = new SkuArgs { Name = "Basic" },
            AdminUserEnabled = true,
        });

        var credentials = Output.Tuple(resourceGroup.Name, Registry.Name).Apply(items =>
            ListRegistryCredentials.InvokeAsync(new ListRegistryCredentialsArgs
            {
                ResourceGroupName = items.Item1,
                RegistryName = items.Item2
            }));
        var adminUsername = credentials.Apply(credentials => credentials.Username!);
        var adminPassword = credentials.Apply(credentials => credentials.Passwords[0].Value!);

        var myImage = new Image("test", new ImageArgs
        {
            ImageName = Output.Format($"{Registry.LoginServer}/test:{ImageVersion}"),
            Build = new DockerBuildArgs
            {
                Context = "../../",
                Platform = "linux/amd64",
            },
            Registry = new ImageRegistryArgs
            {
                Server = Registry.LoginServer,
                Username = adminUsername,
                Password = adminPassword
            }
        });
```

### Output of `pulumi about`

```
running 'dotnet build -nologo .'
  Determining projects to restore...

  Restored /home/raniz/src/[...].csproj (in 437 ms).

  [...] -> /home/raniz/src/[...].dll



Build succeeded.

    0 Warning(s)
    0 Error(s)


Time Elapsed 00:00:05.12

'dotnet build -nologo .' completed successfully
CLI
Version      3.134.1
Go Version   go1.23.1
Go Compiler  gc

Plugins
KIND      NAME          VERSION
resource  azure-native  2.64.0
resource  docker        4.5.6
language  dotnet        unknown

Host
OS       arch
Version
Arch     x86_64

This project is written in dotnet: executable='/usr/bin/dotnet' version='8.0.108'

Current Stack: organization/[...]/dev

TYPE                                        URN
pulumi:pulumi:Stack                         urn:pulumi:dev::[...]::pulumi:pulumi:Stack::[...]-dev
custom:stack:WebApp                         urn:pulumi:dev::[...]::custom:stack:WebApp::webapp
pulumi:providers:azure-native               urn:pulumi:dev::[...]::pulumi:providers:azure-native::default_2_63_0
azure-native:resources:ResourceGroup        urn:pulumi:dev::[...]::azure-native:resources:ResourceGroup::ResourceGroup
azure-native:operationalinsights:Workspace  urn:pulumi:dev::[...]::azure-native:operationalinsights:Workspace::loganalytics
azure-native:containerregistry:Registry     urn:pulumi:dev::[...]::azure-native:containerregistry:Registry::registry
azure-native:containerregistry:Task         urn:pulumi:dev::[...]::azure-native:containerregistry:Task::purgeRegistryTask
pulumi:providers:docker                     urn:pulumi:dev::[...]::pulumi:providers:docker::default_4_5_6
azure-native:app:ManagedEnvironment         urn:pulumi:dev::[...]::azure-native:app:ManagedEnvironment::env
pulumi:providers:azure-native               urn:pulumi:dev::[...]::pulumi:providers:azure-native::default_2_64_0
pulumi:providers:docker                     urn:pulumi:dev::[...]::pulumi:providers:docker::default_4_5_5
docker:index/image:Image                    urn:pulumi:dev::[...]::docker:index/image:Image::test
azure-native:app:ContainerApp               urn:pulumi:dev::[...]::azure-native:app:ContainerApp::app


Found no pending operations associated with dev

Backend
Name           futur
URL            azblob://pulumi-state?storage_account=[...]
User           raniz
Organizations
Token type     personal

Dependencies:
NAME                VERSION
GitVersion.MsBuild  6.0.2
Pulumi              3.67.1
Pulumi.AzureNative  2.64.0
Pulumi.Docker       4.5.6

Pulumi locates its logs in /tmp by default
```

### Additional context

```
error: Running program '<path>.dll' failed with an unhandled exception:
Grpc.Core.RpcException: Status(StatusCode="Unknown", Detail="invocation of azure-native:containerregistry:listRegistryCredentials returned an error: request failed /subscriptions/<subscription>/resourceGroups/<resourceGroup>/providers/Microsoft.ContainerRegistry/registries/<registry>/listCredentials: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: parameter 'resource' cannot be empty")
   at async Task<InvokeResponse> Pulumi.GrpcMonitor.InvokeAsync(ResourceInvokeRequest request)
   at async Task<SerializationResult> Pulumi.Deployment.InvokeRawAsync(string token, SerializationResult argsSerializationResult, InvokeOptions options, RegisterPackageRequest registerPackageRequest) x 2
   at async Task<T> Pulumi.Deployment.InvokeAsync<T>(string token, InvokeArgs args, InvokeOptions options, bool convertResult, RegisterPackageRequest registerPackageRequest)
   at async Task<OutputData<T>> Pulumi.Output<T>+<>c__DisplayClass12_0.<Create>g__GetData|0(?)+GetData(?)
   at async Task<OutputData<U>> Pulumi.Output<T>.ApplyHelperAsync<U>(Task<OutputData<T>> dataTask, Func<T, Output<U>> func) x 3
   at async Task<OutputData<object>> Pulumi.Output<T>.Pulumi.IOutput.GetDataAsync()
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 2
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeDictionaryAsync(string ctx, IDictionary dictionary, bool keepResources, bool keepOutputValues)
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeInputArgsAsync(string ctx, InputArgs args, bool keepResources, bool keepOutputValues)
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 3
   at async Task<RawSerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesRawAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<SerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<PrepareResult> Pulumi.Deployment.PrepareResourceAsync(string label, Resource res, bool custom, bool remote, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies, Result result)> Pulumi.Deployment.RegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies, Result result)> Pulumi.Deployment.ReadOrRegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task Pulumi.Deployment.CompleteResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, ImmutableDictionary<string, IOutputCompletionSource> completionSources, RegisterPackageRequest registerPackageRequest)
   at async Task<OutputData<object>> Pulumi.Output<T>.Pulumi.IOutput.GetDataAsync()
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 2
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeDictionaryAsync(string ctx, IDictionary dictionary, bool keepResources, bool keepOutputValues)
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeInputArgsAsync(string ctx, InputArgs args, bool keepResources, bool keepOutputValues)
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues)
   at async Task<ImmutableArray<object>> Pulumi.Serialization.Serializer.SerializeListAsync(string ctx, IList list, bool keepResources, bool keepOutputValues)
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 3
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeDictionaryAsync(string ctx, IDictionary dictionary, bool keepResources, bool keepOutputValues)
   at async Task<ImmutableDictionary<string, object>> Pulumi.Serialization.Serializer.SerializeInputArgsAsync(string ctx, InputArgs args, bool keepResources, bool keepOutputValues)
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues) x 3
   at async Task<RawSerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesRawAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<SerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<PrepareResult> Pulumi.Deployment.PrepareResourceAsync(string label, Resource res, bool custom, bool remote, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies, Result result)> Pulumi.Deployment.RegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task<(string urn, string id, Struct data, ImmutableDictionary<string, ImmutableHashSet<Resource>> dependencies, Result result)> Pulumi.Deployment.ReadOrRegisterResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, RegisterPackageRequest registerPackageRequest)
   at async Task Pulumi.Deployment.CompleteResourceAsync(Resource resource, bool remote, Func<string, Resource> newDependency, ResourceArgs args, ResourceOptions options, ImmutableDictionary<string, IOutputCompletionSource> completionSources, RegisterPackageRequest registerPackageRequest)
   at async Task<T> Pulumi.Output<T>.GetValueAsync(T whenUnknown)
   at async Task<string> Pulumi.Deployment+EngineLogger.TryGetResourceUrnAsync(Resource resource)
   at async Task<OutputData<U>> Pulumi.Output<T>.ApplyHelperAsync<U>(Task<OutputData<T>> dataTask, Func<T, Output<U>> func) x 2
   at async Task<OutputData<object>> Pulumi.Output<T>.Pulumi.IOutput.GetDataAsync()
   at async Task<OutputData<ImmutableArray<T>>> Pulumi.Output<T>.AllHelperAsync(ImmutableArray<Input<T>> inputs)
   at async Task<OutputData<U>> Pulumi.Output<T>.ApplyHelperAsync<U>(Task<OutputData<T>> dataTask, Func<T, Output<U>> func)
   at async Task<OutputData<object>> Pulumi.Output<T>.Pulumi.IOutput.GetDataAsync()
   at async Task<object> Pulumi.Serialization.Serializer.SerializeAsync(string ctx, object prop, bool keepResources, bool keepOutputValues)
   at async Task<RawSerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesRawAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<SerializationResult> Pulumi.Deployment.SerializeFilteredPropertiesAsync(string label, IDictionary<string, object> args, Predicate<string> acceptKey, bool keepResources, bool keepOutputValues)
   at async Task<Struct> Pulumi.Deployment.SerializeAllPropertiesAsync(string label, IDictionary<string, object> args, bool keepResources, bool keepOutputValues)
   at async Task Pulumi.Deployment.RegisterResourceOutputsAsync(Resource resource, Output<IDictionary<string, object>> outputs)
   at async Task<T> Pulumi.Output<T>.GetValueAsync(T whenUnknown)
   at async Task<string> Pulumi.Deployment+EngineLogger.TryGetResourceUrnAsync(Resource resource)
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Hi @Raniz85, this issue was fixed in release 2.64.1. We apologize for the trouble.

**Comment by Raniz85** [comment]
Ah, great. That wasn't available when I opened the ticket. I just checked, and it works with 2.64.1. Thanks!
